src/coredump/coredump.c

```diff
@@ -1023,91 +1039,93 @@ static int send_iovec(const struct iovec_wrapper *iovw, int input_fd) {
+                r = iovw_put_string_field(iovw, meta_field_names[i], t);
+                if (r < 0)
+                        return r;
         }
 
-        if (cg_pid_get_user_unit(pid, &t) >= 0)
-                iovw_put_string_field_free(iovw, "COREDUMP_USER_UNIT=", t);
-
-        /* The next few are mandatory */
-        r = iovw_put_string_field(iovw, "COREDUMP_PID=", context[CONTEXT_PID]);
-        if (r < 0)
-                return r;
+        /* Cache some of the process metadata we collected so far and that we'll need to
+         * access soon */
+        return cache_process_info(cache, iovw);
+}
 
-        r = iovw_put_string_field(iovw, "COREDUMP_UID=", context[CONTEXT_UID]);
-        if (r < 0)
-                return r;
+static int gather_pid_metadata_from_runtime(struct iovec_wrapper *iovw, ProcessInfoCache *cache) {
+        uid_t owner_uid;
+        pid_t pid;
+        char *t;
+        const char *p;
+        int r;
 
-        r = iovw_put_string_field(iovw, "COREDUMP_GID=", context[CONTEXT_GID]);
-        if (r < 0)
-                return r;
+        /* Note that if we fail on oom later on, we do not roll-back changes to the iovec
+         * structure. (It remains valid, with the first iovec fields initialized.) */
 
-        r = iovw_put_string_field(iovw, "COREDUMP_SIGNAL=", context[CONTEXT_SIGNAL]);
-        if (r < 0)
-                return r;
+        pid = cache->pid;
 
-        r = iovw_put_string_field(iovw, "COREDUMP_RLIMIT=", context[CONTEXT_RLIMIT]);
+        /* The following is mandatory */
+        r = get_process_comm(pid, &t);
         if (r < 0)
-                return r;
+                return log_error_errno(r, "Failed to get COMM: %m");
 
-        r = iovw_put_string_field(iovw, "COREDUMP_HOSTNAME=", context[CONTEXT_HOSTNAME]);
+        r = iovw_put_string_field_free(iovw, "COREDUMP_COMM=", t);
         if (r < 0)
                 return r;
 
-        r = iovw_put_string_field(iovw, "COREDUMP_COMM=", context[CONTEXT_COMM]);
-        if (r < 0)
-                return r;
+        /* The following are optional but we used them if present */
+        if (get_process_exe(pid, &t) >= 0)
+                iovw_put_string_field_free(iovw, "COREDUMP_EXE=", t);
+        else
+                log_warning_errno(r, "Failed to get EXE, ignoring: %m");
 
-        if (context[CONTEXT_EXE]) {
-                r = iovw_put_string_field(iovw, "COREDUMP_EXE=", context[CONTEXT_EXE]);
-                if (r < 0)
-                        return r;
-        }
+        if (cg_pid_get_unit(pid, &t) >= 0)
+                iovw_put_string_field_free(iovw, "COREDUMP_UNIT=", t);
 
         /* The next are optional */
+        if (cg_pid_get_user_unit(pid, &t) >= 0)
+                iovw_put_string_field_free(iovw, "COREDUMP_USER_UNIT=", t);
+
         if (sd_pid_get_session(pid, &t) >= 0)
-                (void) iovw_put_string_field_free(iovw, "COREDUMP_SESSION=", t);
```

Comment:
Why drop the `(void)`? They are useful to tell the reader (and static analyzer) that we purposefully ignore failure.

Reply:
@keszybz is right. Coverity will definitely complain about that. Those "defects" usually look like https://scan6.coverity.com/reports.htm#v36250/p10110/fileInstanceId=90975007&defectInstanceId=27764895&mergedDefectId=1402375.

Speaking of Coverity, @fbuihuu if you're interested, I (or anybody else who has access) can invite you to the systemd organization on Coverity Scan.

Reply:
Indeed I'll restore them.

Reply:
@evverx yes please, I'm interested.

Reply:
@fbuihuu I sent an invitation to the email address from your github profile. It should work but if doesn't just let me know. Regarding the defect density, it went from 0.06 to 0.28 after Coverity was upgraded recently. Usually it's not that bad.